### PR TITLE
 find by numeric code

### PR DESCRIPTION
**Pull request by Stocco**
Created find Currency by iso-4217 numeric code.

Refer to iso codes:
https://en.wikipedia.org/wiki/ISO_4217

### CODE REVIEW

File: currency.go [hunk 1]
```diff
@@ -211,6 +229,21 @@ func GetCurrency(code string) *Currency {
 	return currencies[code]
 }
 
+// GetCurrencyByNumericCode returns the currency given the numeric code defined in ISO-4271.
```
Comment (r1cm3d) [inline currency.go]:
```suggestion
// CurrencyByNumericCode returns the currency given the numeric code defined in ISO-4271.
```

File: currency.go [hunk 2]
```diff
@@ -211,6 +229,21 @@ func GetCurrency(code string) *Currency {
...
+func GetCurrencyByNumericCode(code string) (*Currency, error) {
```
Comment (r1cm3d) [inline currency.go]:
```suggestion
func CurrencyByNumericCode(code string) (*Currency, error) {
```

File: currency.go [hunk 3]
```diff
@@ -211,6 +229,21 @@ func GetCurrency(code string) *Currency {
...
+	if code == "" {
+		return nil, ErrCurrencyNotFound
```
Comment (r1cm3d) [inline currency.go]: IMHO, this error is unnecessary. If there is no Currency, just return `nil`

**Review by r1cm3d — CHANGES_REQUESTED** [review]

File: currency.go (hunk 3 above)
Comment (r1cm3d) [reply currency.go]: Or prefer ["comma, ok"](https://golang.org/doc/effective_go#maps) idiom

**Review by Guilherme-Chaddad — APPROVED** [review]

**Review by r1cm3d — APPROVED** [review]
LGTM

**Pull request closed without merging**